sdk/src/main/java/com/google/cloud/dataflow/sdk/util/WatermarkHold.java

```diff
@@ -247,6 +250,8 @@ private Instant addElementHold(ReduceFn<?, ?, ?, W>.ProcessValueContext context)
     } else {
       which = "on time";
       tooLate = false;
+      Preconditions.checkState(!elementHold.isAfter(BoundedWindow.TIMESTAMP_MAX_VALUE),
```

Comment:
nit: I think it may be better to follow the more typical +4 indentation used elsewhere in this file.